packages/state/source/repository.ts

```diff
@@ -111,6 +111,48 @@ export class Repository implements Contracts.State.Repository {
 		}
 	}
 
+	public changesToJson(): Contracts.State.RepositoryChange {
+		const set = {};
+		const forget: string[] = [];
+
+		for (const name of this.#setAttributes) {
+			if (this.#setAttributes.has(name)) {
+				continue;
+			}
+
+			forget.push(name);
+		}
+
+		for (const name of this.#setAttributes) {
+			set[name] = this.getAttributeHolder(name).toJson();
+		}
```

Comment:
for-loops can be unified

Reply:
Actually first should loop trough forget attributes. Fixed.